src/types.ts

```diff
@@ -62,6 +66,13 @@ type Prettify<T> = {
 
 type TupleToUnion<T extends unknown[]> = T[number]
 
+type TupleToIntersection<
+  T extends unknown[],
+  output extends unknown = unknown,
+> = T extends [infer first, ...infer rest]
+  ? TupleToIntersection<rest, output & first>
+  : Prettify<output>
+
```

Comment:
How is this different from `MergeObjs` ?

Reply:
The result of `TupleToIntersection<[number, string]>` is `never`. `MergeObjs` currently assumes the input types always have keys.